src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoNCreationUtils.java

```diff
@@ -31,298 +30,198 @@
-        final RealMatrix logNormalsPseudoInverse = logNormalsSVD.getPinv();
-
-        logger.info("Calculating the reduced PoN inverse matrix...");
-        final long riStartTime = System.currentTimeMillis();
-        final RealMatrix reducedCountsPseudoInverse = SVDFactory.createSVD(reducedCounts, ctx).getPinv();
-        final long riEndTime = System.currentTimeMillis();
-        logger.info(String.format("Finished calculating the reduced PoN inverse matrix. Elapse of %d seconds", (riEndTime - riStartTime) / 1000));
-        return new ReductionResult(logNormalsPseudoInverse, reducedCounts, reducedCountsPseudoInverse, logNormalsSVD.getSingularValues());
     }
 
     /**
-     * Calculate the number of eigen sample given the user preferences and the
-     * result of the log-normals SVD.
+     * Subsets targets in the input count to the usable ones based on the percentile threshold indicated
+     * by the user.
      *
-     * @param requestedNumberOfEigenSamples the user requested eigen values (empty means the user didn't specify any in particular).
-     * @param numberOfCountColumns number of count columns in the original input.
-     * @param logNormalsSVD SVD results on the log-normal counts.
-     * @return always greater than 0.
+     * <p>
+     *     It returns a pair of object, where the left one is the updated read-counts with only the usable
+     *     targets, and the right one is the corresponding target factors.
+     * </p>
+     *
+     * @param readCounts the input read-counts.
+     * @param targetFactorPercentileThreshold the minimum median count percentile under which targets are not considered useful.
+     * @return never {@code null}.
      */
     @VisibleForTesting
-    static int determineNumberOfEigenSamples(final OptionalInt requestedNumberOfEigenSamples, final int numberOfCountColumns, final SVD logNormalsSVD, final Logger logger) {
-        final int numberOfEigenSamples;
-        if (requestedNumberOfEigenSamples.isPresent()) {
-            if (requestedNumberOfEigenSamples.getAsInt() > numberOfCountColumns) {
-                logger.warn(String.format("The number of requested eigen samples (%d) is larger than the available number of read count columns after filtering (%d), thus we will have to use the latter.", requestedNumberOfEigenSamples.getAsInt(), numberOfCountColumns));
-            }
-            numberOfEigenSamples = Math.min(requestedNumberOfEigenSamples.getAsInt(), numberOfCountColumns);
+    static Pair<ReadCountCollection, double[]> subsetReadCountsToUsableTargets(final ReadCountCollection readCounts,
+                                                                               final double targetFactorPercentileThreshold, final Logger logger) {
+        final double[] targetFactors = calculateTargetFactors(readCounts);
+        final Percentile percentile = new Percentile(targetFactorPercentileThreshold);
+        final double threshold = percentile.evaluate(targetFactors);
```

Comment:
I would find it clearer if `threshold` were on the LHS and `targetFactorPercentileThreshold` on the RHS via `final double threshold = new Percentile(targetFactorPercentileThreshold).evaluate(targetFactors);`